notification-service/src/main/java/com/redhat/parodos/notification/controller/advice/ControllerExceptionHandler.java

```diff
@@ -40,7 +40,7 @@ public ErrorMessageDTO handleConstraintViolationException(ConstraintViolationExc
 	@ExceptionHandler(value = { UsernameNotFoundException.class, NotificationRecordNotFoundException.class })
 	@ResponseBody
 	@ResponseStatus(value = HttpStatus.NOT_FOUND)
-	public ErrorMessageDTO resourceNotFoundException(UsernameNotFoundException ex, WebRequest request) {
+	public ErrorMessageDTO resourceNotFoundException(RuntimeException ex, WebRequest request) {
```

Comment:
Why runtime exception?

Reply:
It should be of a type that can accept the types defined in the 
```
@ExceptionHandler(value = { UsernameNotFoundException.class, NotificationRecordNotFoundException.class })
```

Reply:
It may fail at runtime if an unsupported type is thrown. No warning from the compiler...